### PR TITLE
Added cloudflare turnstile

### PR DESCRIPTION
**Pull request by ravirajput10**
Fixed by: #375

### CODE REVIEW

File: packages/utils/src/validate-turnstile-token.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const SECRET_KEY = process.env.TURNSTILE_SECRET_KEY;
```
Comment (rajat1saxena) [inline packages/utils/src/validate-turnstile-token.ts:1]: Move this file's code to `apps/web/app/api/cloudflare` route handler as a private function.

File: packages/utils/src/validate-turnstile-token.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        return true;
+    }
+    if (
+        outcome["error-codes"] &&
+        outcome["error-codes"][0] === "timeout-or-duplicate"
+    ) {
+        return true;
+    }
+    // eslint-disable-next-line no-console
+    console.log(`Turnstile validation failed`, outcome);
```
Comment (rajat1saxena) [inline packages/utils/src/validate-turnstile-token.ts:29]: Log this to the database `Log` collection as an error using Logger (from `/apps/web/services/logger.ts`) utility.

**Review by rajat1saxena — CHANGES_REQUESTED** [review]

File: packages/utils/src/validate-turnstile-token.ts (hunk 1 above)
Comment (ravirajput10) [reply packages/utils/src/validate-turnstile-token.ts]: Moved the code

File: packages/utils/src/validate-turnstile-token.ts (hunk 2 above)
Comment (ravirajput10) [reply packages/utils/src/validate-turnstile-token.ts]: Logged the error in the DB's log collection

**Comment by rajat1saxena** [comment]
@ravirajput10 The code quality checks are failing. Please check.

**Pull request merged**